controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -555,148 +563,208 @@ def test_create_volume_with_zero_size(self, storage_agent, array_type):
         self.mediator.create_volume = Mock()
-        self.mediator.create_volume.return_value = utils.get_mock_mediator_response_volume(10, vol_name, "wwn2", "a9k")
-        self.mediator.get_snapshot_by_id = Mock()
-        self.mediator.get_snapshot_by_id.return_value = utils.get_mock_mediator_response_snapshot(snap_capacity_bytes,
-                                                                                                  snap_name,
-                                                                                                  snapshot_id, vol_name,
-                                                                                                  "a9k")
-        self.mediator.copy_to_existing_volume_from_snapshot = Mock()
-        self.servicer.CreateVolume(self.request, self.context)
+        self.mediator.create_volume.return_value = utils.get_mock_mediator_response_volume(10, volume_name, "wwn2",
+                                                                                           "a9k")
+        self.mediator.get_object_by_id = Mock()
+        self.mediator.get_object_by_id.return_value = utils.get_mock_mediator_response_snapshot(snapshot_capacity_bytes,
+                                                                                                snapshot_name,
+                                                                                                snapshot_id,
+                                                                                                volume_name,
+                                                                                                "a9k")
+        self.mediator.copy_to_existing_volume_from_source = Mock()
+        response_volume = self.servicer.CreateVolume(self.request, self.context)
         self.assertEqual(self.context.code, grpc.StatusCode.OK)
-        self.mediator.copy_to_existing_volume_from_snapshot.assert_called_once_with(vol_name, snap_name,
-                                                                                    snap_capacity_bytes,
-                                                                                    self.capacity_bytes,
-                                                                                    'pool1')
+        self.mediator.copy_to_existing_volume_from_source.assert_called_once_with(volume_name, snapshot_name,
+                                                                                  snapshot_capacity_bytes,
+                                                                                  self.capacity_bytes,
+                                                                                  'pool1')
+        self.assertEqual(response_volume.volume.content_source.volume.volume_id, '')
+        self.assertEqual(response_volume.volume.content_source.snapshot.snapshot_id, snapshot_id)
+
+    # @patch("controller.controller_server.csi_controller_server.detect_array_type")

```

Comment:
?

Reply:
this id the idempotent test, they are true negative so they ae bad but I didnt want to remove them because they only need fixes that will be happening in CSI-1540

Reply:
we have git history. no need to merge commented out code, ever.

Reply:
done (moved to CSI-1540)